### DISCUSSION TITLE
XmlMapper serializes jsonAppend property twice

### DISCUSSION THREAD

**Discussion by stepince** [FasterXML/jackson-databind#3806] [Q&A] [answered]
XmlMapper is serializing jsonAppend virtual property twice.  ObjectMapper for json works correctly.

jackson version: 2.14.1

`
public class VirtualBeanPropertyWriterTest {
    @Test
    public void testJsonAppend() throws Exception {
        ObjectMapper mapper = new XmlMapper();
        String xml = mapper.writeValueAsString(new Pojo("foo"));
        assertEquals("<Pojo><name>foo</name><virtual>bar</virtual></Pojo>",xml);
    }

    @JsonAppend(props = @JsonAppend.Prop(name = "virtual", value = MyVirtualPropertyWriter.class))
    public static class Pojo {
        private final String name;

        public Pojo(String name) {
            this.name = name;
        }
        public String getName() {
            return name;
        }
    }

    public static class MyVirtualPropertyWriter extends VirtualBeanPropertyWriter {
        public MyVirtualPropertyWriter() {}

        protected MyVirtualPropertyWriter(BeanPropertyDefinition propDef, Annotations contextAnnotations,
                                          JavaType declaredType) {
            super(propDef, contextAnnotations, declaredType);
        }

        @Override
        protected Object value(Object bean, JsonGenerator jgen, SerializerProvider prov) throws Exception {
            return "bar";
        }

        @Override
        public VirtualBeanPropertyWriter withConfig(MapperConfig<?> config, AnnotatedClass declaringClass,
                                                    BeanPropertyDefinition propDef, JavaType type) {

            return new MyVirtualPropertyWriter(propDef, declaringClass.getAnnotations(), type);
        }
    }
}
`

output
org.opentest4j.AssertionFailedError: 
Expected :`<Pojo><name>foo</name><virtual>bar</virtual></Pojo>`
Actual   :`<Pojo><name>foo</name><virtual>bar</virtual><virtual>bar</virtual></Pojo>`

**Comment by cowtowncoder** [comment, marked as answer]
Why discussion? This sounds like a bug report. Also, wrong repo; will transfer, then convert.

**EDIT**: created issue https://github.com/FasterXML/jackson-dataformat-xml/issues/578 for follow up.

**Reply by stepince** [reply]
Hi,  
     I just wanted to verify it was a bug.  I wanted to see if there is a work around.
Steve

**Reply by cowtowncoder** [reply]
It definitely sounds like one, hence bug report. It is ok to file issues even if not 100% certain something is a flaw, if it seems it could well be.
Thanks!
